diff --git a/src/platform/linux/backtrace.cpp b/src/platform/linux/backtrace.cpp
index d2a08c0..80a7092 100644
--- a/src/platform/linux/backtrace.cpp
+++ b/src/platform/linux/backtrace.cpp
@@ -3,3 +3,4 @@
+#include <cstdint>
 #include <cstdlib>    // for free
 #include <cxxabi.h>   // for abi::__cxa_demangle
 #include <execinfo.h> // for backtrace, backtrace_symbols
diff --git a/src/utility/argument_parsing/argument.cpp b/src/utility/argument_parsing/argument.cpp
index cee542f..fa196fc 100644
--- a/src/utility/argument_parsing/argument.cpp
+++ b/src/utility/argument_parsing/argument.cpp
@@ -3,4 +3,5 @@
 #include <cstdlib>
+#include <cstring>
 #include <cxxabi.h>
 #include <iostream>
 #include <memory>
diff --git a/src/platform/linux/singleton_process.cpp b/src/platform/linux/singleton_process.cpp
index b9c00dc..c4a8364 100644
--- a/src/platform/linux/singleton_process.cpp
+++ b/src/platform/linux/singleton_process.cpp
@@ -3,3 +3,4 @@
 #include <cerrno>
 #include <chrono>
+#include <cstring>
 #include <fcntl.h> /* For O_* constants */
